examples/cmri_mpca/config.py

```diff
@@ -0,0 +1,77 @@
+"""
+Default configurations for CMR processing and classification
+"""
+
+import os
+
+import numpy as np
+import pydicom
+from yacs.config import CfgNode as CN
+
+# -----------------------------------------------------------------------------
+# Config definition
+# -----------------------------------------------------------------------------
+
+_C = CN()
+
+# -----------------------------------------------------------------------------
+# Dataset
+# -----------------------------------------------------------------------------
+_C.DATASET = CN()
+_C.DATASET.SOURCE = "https://github.com/pykale/data/raw/main/image_data/ShefPAH-179/SA_64x64.zip"
+_C.DATASET.ROOT = "../data"  # '/shared/tale2/Shared'
+_C.DATASET.IMG_DIR = "DICOM"
+_C.DATASET.FILENAME = "PAH_SA_64x64"
+_C.DATASET.FILE_FORAMT = "zip"
+_C.DATASET.LANDMARK_FILE = "landmarks.csv"
+_C.DATASET.MASK_FILE = "Mask/phase1.dcm"
+# ---------------------------------------------------------------------------- #
+# Image processing
+# ---------------------------------------------------------------------------- #
+_C.PROC = CN()
+_C.PROC.SCALE = 2
+
+# ---------------------------------------------------------------------------- #
+# MPCA Pipeline
+# ---------------------------------------------------------------------------- #
+_C.MPCA = CN()
+_C.MPCA.CLF = ""  # choices=['CDAN', 'CDAN-E', 'DANN']
+
+# ---------------------------------------------------------------------------- #
+# Misc options
+# ---------------------------------------------------------------------------- #
+_C.OUTPUT = CN()
+_C.OUTPUT.ROOT = "./outputs"  # output_dir
+
+
+def get_cfg_defaults():
+    return _C.clone()
+
+
+def read_dicom_imgs(dicom_path):
+    sub_dirs = os.listdir(dicom_path)
+    all_ds = []
+    sub_ids = []
+    for sub_dir in sub_dirs:
+        sub_ds = []
+        sub_path = os.path.join(dicom_path, sub_dir)
+        phase_files = os.listdir(sub_path)
+        for phase_file in phase_files:
+            dataset = pydicom.dcmread(os.path.join(sub_path, phase_file))
+            sub_ds.append(dataset)
+        sub_ds.sort(key=lambda x: x.InstanceNumber, reverse=False)
+        sub_ids.append(int(sub_ds[0].PatientID))
```

Comment:
Subject should be better than Patient. What does `sub` in `sub_ids` stand for?

Reply:
PatientID is one of the meta information of DICOM. `sub_ds` is a list of dicom images, i.e. data for one subject. `sub_ids` is the indices for subjects. Here we sort the IDs for subjects to make the order of subject indices consistent with the order in the .csv file (landmarks and labels).